pkg/linkValidator/linkValidator_test.go

```diff
@@ -0,0 +1,270 @@
+		},
+		{
+			name:     "multiple URLs",
+			input:    "Visit https://example.com and http://test.com for more info",
+			expected: []string{"https://example.com", "http://test.com"},
+		},
+		{
+			name:     "URL with trailing punctuation",
+			input:    "See https://example.com.",
+			expected: []string{"https://example.com"},
+		},
+		{
+			name:     "no URLs",
+			input:    "This is just text without any URLs",
+			expected: []string{},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			result := extractURLs(tc.input)
+			if len(result) != len(tc.expected) {
+				t.Errorf("Expected %d URLs, got %d: %v", len(tc.expected), len(result), result)
+				return
+			}
+
+			for i, url := range result {
+				if url != tc.expected[i] {
+					t.Errorf("Expected URL %s, got %s", tc.expected[i], url)
+				}
+			}
+		})
+	}
+}
+
+func TestLinkValidator_checkURL(t *testing.T) {
+	// Create a test server that returns 200 OK
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer server.Close()
+
+	// Create a test server that returns 404
+	server404 := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusNotFound)
+	}))
+	defer server404.Close()
+
+	// Create a test server that returns 500
+	server500 := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusInternalServerError)
+	}))
+	defer server500.Close()
+
+	lv := NewLinkValidator()
+
+	testCases := []struct {
+		name           string
+		url            string
+		expectedStatus int
+		expectError    bool
+	}{
+		{
+			name:           "valid URL",
+			url:            server.URL,
+			expectedStatus: 200,
+			expectError:    false,
+		},
+		{
+			name:           "404 URL",
+			url:            server404.URL,
+			expectedStatus: 404,
+			expectError:    false,
+		},
+		{
+			name:           "500 URL",
+			url:            server500.URL,
+			expectedStatus: 500,
+			expectError:    false,
+		},
+		{
+			name:           "invalid URL",
+			url:            "not-a-valid-url",
+			expectedStatus: 0,
+			expectError:    true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			statusCode, err := lv.checkURL(tc.url)
+			if tc.expectError && err == nil {
+				t.Error("Expected error but got none")
+			}
+			if !tc.expectError && err != nil {
+				t.Errorf("Expected no error but got: %v", err)
+			}
+			if !tc.expectError && statusCode != tc.expectedStatus {
+				t.Errorf("Expected status code %d, got %d", tc.expectedStatus, statusCode)
+			}
+		})
+	}
+}
+
+func TestLinkValidator_ValidateLinks(t *testing.T) {
+	// Create a test server for valid URLs (200 OK)
+	serverOK := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer serverOK.Close()
```

Comment:
There's a lot of duplication here. It doesn't matter too much, but one option here is to use a private function to generate test servers for a given http status. For example (pseudo code)
```go
fun newTestServer(status http.Status) ... { }
```